github-actions/branch-manager/lib/main.ts

```diff
@@ -62,35 +66,57 @@ async function run() {
         ({name}) => name === actionLabels.ACTION_MERGE.name,
       );
       if (hasMergeLabel) {
-        await createWorkflowForPullRequest();
+        await createWorkflowForPullRequest({
+          repo: context.issue.repo,
+          owner: context.issue.owner,
+          pr: `${context.issue.number}`,
+          sha: payload.pull_request.head.sha,
+        });
       }
     }
 
     if (context.payload.action === 'labeled') {
-      const event = context.payload as PullRequestLabeledEvent;
-      // If the merge label has been added, or if target labels have changed,
-      // another update of the merge status is needed.
+      const payload = context.payload as PullRequestLabeledEvent;
+      // If the merge label has been added, or if target labels have changed // another update of
+      // the merge status is needed.
       if (
-        event.label.name === actionLabels.ACTION_MERGE.name ||
-        targetLabelNames.has(event.label.name)
+        payload.label.name === actionLabels.ACTION_MERGE.name ||
+        targetLabelNames.has(payload.label.name)
       ) {
-        await createWorkflowForPullRequest();
+        await createWorkflowForPullRequest({
+          repo: context.issue.repo,
+          owner: context.issue.owner,
+          pr: `${context.issue.number}`,
+          sha: payload.pull_request.head.sha,
+        });
       }
     }
   }
 }
 
-/** The pull request from the context of the action being run, used as the default pull request. */
-const pullRequestFromContext = {
-  repo: context.issue.repo,
-  owner: context.issue.owner,
-  pr: `${context.issue.number}`,
+type WorkflowInputs = {
+  repo: string;
+  owner: string;
+  pr: string;
+  sha?: string;
 };
-type WorkflowInputs = typeof pullRequestFromContext;
 
 /** Create a workflow dispatch event to trigger the pr to be evaluated for mergeability. */
-function createWorkflowForPullRequest(prInfo?: Partial<WorkflowInputs>) {
-  const inputs = {...pullRequestFromContext, ...prInfo};
+async function createWorkflowForPullRequest(inputs: WorkflowInputs) {
+  if (inputs.sha !== undefined) {
+    console.info(`Setting pending status for pr #${inputs.pr}`);
+    await github().then((api) =>
+      api.repos.createCommitStatus({
+        repo: inputs.repo,
+        owner: inputs.owner,
+        state: 'pending',
+        description: 'Running mergibility check',
+        sha: inputs.sha!,
```

Comment:
Is this needed?

Reply:
Unfortunately yes, I think since the `inputs` is used in the closure, the `!== undefined` check that happens outside of the closure does not come into the typings in the closure.

I refactored to avoid the closure, but its definitely an odd case I hadn't run into before.